### PR TITLE
feat: fetch _all_ channels from status.nixos.org

### PR DESCRIPTION
**Pull request by RaitoBezarius**
We use monitoring.nixos.org's Prometheus instance as source of truth. It is supposed to run regularly to update the channel view all the time.

The infrastructure counterpart (a timer) has to be set.

### CODE REVIEW

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    if name == "unmaintained":
+        return NixChannel.ChannelState.END_OF_LIFE
+    elif name == "deprecated":
+        return NixChannel.ChannelState.DEPRECATED
+    elif name == "beta":
+        return NixChannel.ChannelState.BETA
+    elif name == "stable":
+        return NixChannel.ChannelState.STABLE
+    elif name == "rolling":
+        return NixChannel.ChannelState.UNSTABLE
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:49]: Arguably we could rename that field to `ROLLING`

**Comment by alejandrosame** [comment]
The merge conflict was probably introduced by #115, so it'd require reformatting the file.

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+
+@dataclass
+class MonitoredChannel:
+    name: str
+    revision: str
+    status: str
+
+
+def derive_release_version_from_name(name: str) -> str | None:
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:15]:
```suggestion
def release_from_branch(branch: str) -> str | None:
```

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        return None
+
+    ver = parts[1]
+    if "." not in ver:
+        return None
+
+    return ver
+
+
+def derive_state_from_status(name: str) -> NixChannel.ChannelState:
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:39]:
```suggestion
def state_from_status(status: str) -> NixChannel.ChannelState:
```

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        return NixChannel.ChannelState.BETA
+    elif name == "stable":
+        return NixChannel.ChannelState.STABLE
+    elif name == "rolling":
+        return NixChannel.ChannelState.UNSTABLE
+    else:
+        return NixChannel.ChannelState.STAGING
+
+
+def derive_staging_from_name(name: str) -> str:
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:54]:
```suggestion
def staging_from_branch(branch: str) -> str:
```

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            default="https://github.com/NixOS/nixpkgs",
+        )
+
+    def handle(self, *args, **kwargs):
+        fresh_channels = fetch_from_monitoring()
+        defaults = {"repository": kwargs["repository"]}
+        for channel in fresh_channels.values():
+            channel_branch = channel.name
+            staging_branch = derive_staging_from_name(channel.name)
+            specific_defaults = defaults | {
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:98]:
```suggestion
            branch_info = defaults | {
```

File: src/website/shared/management/commands/fetch_all_channels.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                "staging_branch": staging_branch,
+                "state": derive_state_from_status(channel.status),
+                "head_sha1_commit": channel.revision,
+                "release_version": derive_release_version_from_name(channel.name),
+            }
+            print(specific_defaults, channel)
+            print(
+                NixChannel.objects.update_or_create(
```
Comment (fricklerhandwerk) [inline src/website/shared/management/commands/fetch_all_channels.py:106]: Arguably we could rename this to `Branch` at some point.

**Review by fricklerhandwerk — APPROVED** [review]
Cosmetics for better readability.

**Comment by fricklerhandwerk** [comment]
Added type annotations to satisfy linters, fixed up the naming.

**Pull request merged**